modules/dnn/include/opencv2/dnn/csl/error.hpp

```diff
@@ -0,0 +1,20 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_DNN_CSL_ERROR_HPP
+#define OPENCV_DNN_CSL_ERROR_HPP
+
+#include <opencv2/core.hpp>
+
+namespace cv { namespace dnn { namespace cuda4dnn { namespace csl {
+
+    //! exception class for errors thrown by the CUDA APIs
+    class CUDAException : public cv::Exception {
```

Comment:
`cuBLASException` and `cuDNNException` are derived from `CUDAException`. The CUDA runtime does not have an exception class of its own. Should CUDA runtime have its own exception class? Are the exception classes required?

Reply:
You definitely need some exception class for this stuff, at least `CUDAException`. Beyond that, I think reasonable people can differ about how many subclasses there should be. Mapping generic CUDA runtime errors to `CUDAException` seems good to me.